### ISSUE TITLE
Little Question

### ISSUE THREAD

**Issue by Modius22** [ewsterrenburg/python-otrs#31]
Hi Guys,
Is an expansion planned for the OTRS CMDB?
greetings
Modius

**Comment by mjducharme** [comment]
I plan to do this in a few months after upgrading OTRS. A bug in the OTRS SOAP interface prevented me from even fetching a config item.

**Comment by ewsterrenburg** [comment]
We are not using this functionality in our OTRS (actually, I more or less only use this library to create example soap requests ;), yet I'd be happy to merge. 

@mjducharme 
it seems you are doing more of the work than I am, if you want to become owner of this project, that's fine with me (you should have an invite now as a collaborator)

**Comment by Modius22** [comment]
To explain this perhaps again more precisely =)
I would like to query a CI from the OTRS CMDB from the outside.
As a return I would like to have the CI information as well as the tickets with which the CI is linked.
Do you have an idea?

**Comment by mjducharme** [comment]
Yes, please give me another week and I should have this functionality developed.

**Comment by mjducharme** [comment]
I have made progress but this is still not done. The main challenge is that the XML data for the CI objects has a very dynamic tree structure, so I need to create classes on-the-fly instead of using statically defined classes. This requires some changes to the core code.

**Comment by mjducharme** [comment]
I got the dynamic classes working, but decided it was not the best UI experience for the CIXMLData structure. Since that data structure can have multiple levels of children and is very flexible, I am using a dict for that, and created new functions to recursively create a dict from xml or xml from dict. It is about 80% complete. I have to fix the Attachments feature because for some reason in Config Items the attachment is under the "ConfigItem" structure inside Data instead of in the Data structure, so it doesn't correspond precisely with the way the ticket functions and FAQ functions use the attachments. I need to do a few more fixes to get the attachments to work properly.

**Comment by PromptSec** [comment]
HI @mjducharme !

Great job you is doing.

**Comment by JulianDeal** [comment]
@mjducharme 
This all sounds great! But you didn't push your work, did you? It would be extremely helpful for me right now. Or at least branching it. I don't even need the attachment feature.

**Comment by ewsterrenburg** [comment]
@mjducharme 
Have you looked into this in the mean time?
